tests/pkcli/nersc_test_test.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""Test pkcli.nersc_test
+
+:copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+import pytest
+import pykern.pkio
```

Comment:
We avoid imports at the global scope, because the import may fail, which is better caught in a test.

pytest does support classes for testing. It's better if the two cases are separate entry points that pytest can see, e.g. test_conformance and test_deviance. The words "success" and "failure" are confusing in the context of a test of a test so stick to stricter terms.

The class I think here is overkill but not unreasonably so. I would recommend simplifying and sharing in a different way. For example, exec_cmd could hold the value CmD_BASIC. "result.log" could be part of "check_result" (not here, but could be). Then you could simplify the two cases.